File: beginner_tutorials/scripts/simple_node.py
#!/usr/bin/env python
'''
The above line is must for any python script you write. Infact this line itself is making your script, a python file in linux.
'''
import rospy

count=0
rospy.init_node('simple_node_py') #initialzing the node with name "simple_node_py"
rate = rospy.Rate(10) # 10hz

while not rospy.is_shutdown(): # run the node until Ctrl-C is pressed

	#Write your code here

	rospy.loginfo("simple_node in python is running. count= %d",count) #debug statement
	
	count+=1
	
	rate.sleep()# This makes the loop to iterate at 10Hz rate i.e., 10 times a sec.

File: beginner_tutorials/src/simple_node.cpp
#include <ros/ros.h>

int main(int argc, char** argv)
{
	int count = 0;
	ros::init(argc, argv, "simple_node"); // Node declared with name "simple_node_cpp"
	ros::NodeHandle nh; // NodeHandle will actually initialze the node
	ros::Rate loop_rate(10); // 10Hz
	
	while (ros::ok()) // run the node until Ctrl-C is pressed
	{
		//Wirte your code here
		
		ROS_INFO("simple_node in cpp is running. count= %d",count);// Debug statement
	
		ros::spinOnce(); // spinOnce() is used to call the callbacks (if any) in the code. This is not required for this node But it is a best practise to have it. 
		
		count++;
		
		loop_rate.sleep(); // This makes the loop to iterate at 10Hz i.e., 10 times a sec.
	}
}
